### PR TITLE
VP-26 implement deletion of Velocloud Connection

### PR DESCRIPTION
**Pull request by dherdt**

### CODE REVIEW

**Review by fooker — CHANGES_REQUESTED** [review]
Was there any discussion that I missed? Why not implement a delete method in the API and just call that?

**Review by fooker — CHANGES_REQUESTED** [review]
Why are the changes to `features.xml` necessary?

**Comment by dherdt** [comment]
> Why are the changes to `features.xml` necessary?

otherwise the karaf does not start the plugin. 
I took this commit of https://github.com/OpenNMS/opennms-velocloud-plugin/pull/40/commits/b1896719238e19fee51f5eab8629bf0376fcd928 to test functionality.

**Comment by fooker** [comment]
> > Why are the changes to `features.xml` necessary?
> 
> otherwise the karaf does not start the plugin. I took this commit [b189671](https://github.com/OpenNMS/opennms-velocloud-plugin/commit/b1896719238e19fee51f5eab8629bf0376fcd928) made by [OpenNMS/opennms-velocloud-plugin/commits?author=RangerRick](https://github.com/OpenNMS/opennms-velocloud-plugin/commits?author=RangerRick) to test functionality.

This tells my why you've added the code, but not why it is needed.

There is no dependency from the Client implementations to OIA. The error you've seen must be something different. In addition, client V2 is currently a pure stub and has no deps at all. Therefore sharing the deps with V1 is not the right thing to do.

**Comment by dherdt** [comment]
> > > Why are the changes to `features.xml` necessary?
> > 
> > 
> > otherwise the karaf does not start the plugin. I took this commit [b189671](https://github.com/OpenNMS/opennms-velocloud-plugin/commit/b1896719238e19fee51f5eab8629bf0376fcd928) made by [OpenNMS/opennms-velocloud-plugin/commits?author=RangerRick](https://github.com/OpenNMS/opennms-velocloud-plugin/commits?author=RangerRick) to test functionality.
> 
> This tells my why you've added the code, but not why it is needed.
> 
> There is no dependency from the Client implementations to OIA. The error you've seen must be something different. In addition, client V2 is currently a pure stub and has no deps at all. Therefore sharing the deps with V1 is not the right thing to do.

I have rebased on latest `main` and removed changes in `features.xml`. Everything works fine.

File: plugin/src/main/java/org/opennms/velocloud/connections/ConnectionManager.java [hunk 1]
```diff
@@ -110,6 +109,24 @@ public Connection newConnection(final String alias, final String orchestratorUrl
...
+     * connection with given alias was not found
+     */
+    public boolean deleteConnection(final String alias) {
+        final String vaultAlias = PREFIX + alias.toLowerCase();
+        final var credentials = this.vault.getCredentials(vaultAlias);
+        if (credentials == null) {
+            return false;
+        }
+        this.vault.deleteCredentials(vaultAlias);
+        ConnectionManager.this.clientManager.purgeClient(ConnectionManager.fromStore(credentials));
```
Comment (fooker) [inline plugin/src/main/java/org/opennms/velocloud/connections/ConnectionManager.java:126]:
```suggestion
        this.clientManager.purgeClient(ConnectionManager.fromStore(credentials));
```

**Review by fooker — CHANGES_REQUESTED** [review]
I think we can improve the API a little by having a `Connection#delete()` instead of a `ConnectionManager#delete(alias)`. This would move the handling of none-existing alias to the caller.

**Review by fooker — APPROVED** [review]

**Pull request merged**